wikidata/client.py

```diff
@@ -177,21 +178,30 @@ def decode_datavalue(self,
                       self.datavalue_decoder)
         return decode(self, datatype, datavalue)
 
-    def request(self, path: str) -> Union[
+    def request(self, path: str) -> Optional[Union[
         bool, int, float, str,
         Mapping[
             str,
             Union[bool, int, float, str, Mapping[str, object], Sequence]
         ],
         Sequence[Union[bool, int, float, str, Mapping[str, object], Sequence]]
-    ]:
+    ]]:
         logger = logging.getLogger(__name__ + '.Client.request')
         url = urllib.parse.urljoin(self.base_url, path)
         result = self.cache_policy.get(CacheKey(url))
         if result is None:
             logger.debug('%r: no cache; make a request...', url)
-            response = self.opener.open(url)
-            buffer_ = io.TextIOWrapper(response, encoding='utf-8')
+            try:
+                response = self.opener.open(url)
+            except urllib.error.HTTPError as e:
+                logger.debug('HTTP error code: ', e.code, exc_info=True)
```

Comment:
This will ignore the second argument `e.code` and the log message ends with a single space.  This should be changed like:

```suggestion
                logger.debug('HTTP error code: %s', e.code, exc_info=True)
```